Code/Tools/AssetProcessor/native/unittests/AssetProcessingStateDataUnitTests.cpp

```diff
@@ -1,71 +1,95 @@
-    SourceDatabaseEntryContainer sources;
-    JobDatabaseEntryContainer jobs;
-    ProductDatabaseEntryContainer products;
-    ProductDependencyDatabaseEntryContainer productDependencies;
-    MissingProductDependencyDatabaseEntryContainer missingDependencies;
-
-    QString outName;
-    QString outPlat;
-    QString outJobDescription;
-
-    AZ::Uuid validSourceGuid1 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validSourceGuid2 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validSourceGuid3 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validSourceGuid4 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validSourceGuid5 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validSourceGuid6 = AZ::Uuid::CreateRandom();
-
-    AZ::u32 validFingerprint1 = 1;
-    AZ::u32 validFingerprint2 = 2;
-    AZ::u32 validFingerprint3 = 3;
-    AZ::u32 validFingerprint4 = 4;
-    AZ::u32 validFingerprint5 = 5;
-    AZ::u32 validFingerprint6 = 6;
-
-    AZ::Uuid validBuilderGuid1 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validBuilderGuid2 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validBuilderGuid3 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validBuilderGuid4 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validBuilderGuid5 = AZ::Uuid::CreateRandom();
-    AZ::Uuid validBuilderGuid6 = AZ::Uuid::CreateRandom();
-
-    AZ::Data::AssetType validAssetType1 = AZ::Data::AssetType::CreateRandom();
-    AZ::Data::AssetType validAssetType2 = AZ::Data::AssetType::CreateRandom();
-    AZ::Data::AssetType validAssetType3 = AZ::Data::AssetType::CreateRandom();
-    AZ::Data::AssetType validAssetType4 = AZ::Data::AssetType::CreateRandom();
-    AZ::Data::AssetType validAssetType5 = AZ::Data::AssetType::CreateRandom();
-    AZ::Data::AssetType validAssetType6 = AZ::Data::AssetType::CreateRandom();
-
-    AzToolsFramework::AssetSystem::JobStatus statusQueued = AzToolsFramework::AssetSystem::JobStatus::Queued;
-    AzToolsFramework::AssetSystem::JobStatus statusCompleted = AzToolsFramework::AssetSystem::JobStatus::Completed;
-
-    //////////////////////////////////////////////////////////////////////////
-    //ScanFolder
-    //the database all starts with a scan folder since all sources are have one
-    auto ScanFoldersContainScanFolderID = [](const ScanFolderDatabaseEntryContainer& scanFolders, AZ::s64 scanFolderID) -> bool
+    class AssetProcessingStateDataUnitTest
+        : public UnitTest::AssetProcessorUnitTestBase
+    {
+    protected:
+        void SetUp() override
+        {
+            UnitTest::AssetProcessorUnitTestBase::SetUp();
+
+            EXPECT_TRUE(m_connection.DataExists());
+            EXPECT_TRUE(m_connection.OpenDatabase());
+
+        }
+
+        void TearDown() override
+        {
+            m_connection.CloseDatabase();
+
+            UnitTest::AssetProcessorUnitTestBase::TearDown();
+        }
+
+        AssetProcessor::AssetDatabaseConnection m_connection;
+    };
+
+    TEST_F(AssetProcessingStateDataUnitTest, DataTest_ValidDatabaseConnectionProvided_OperationsSucceed)
```

Comment:
I know you are migrating the tests, but this is a huge unit test. I hope we have plans to break this up a bit as its not maintainable or focused. We should avoid uber unit tests IMHO

Reply:
Sure. I can separate issue to break down this test (as I'm currently doing for the other tests which is even bigger than this one) and jump on it after this PR is merged in since this PR includes the shared base classes that other PRs depend on.